File: app/src/main/java/h_mal/appttude/com/driver/User/forgotPasswordActivity.java
package h_mal.appttude.com.driver.User;

import android.support.annotation.NonNull;
import android.support.v4.app.NavUtils;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;

import h_mal.appttude.com.driver.R;

public class forgotPasswordActivity extends AppCompatActivity {

    String TAG = "forgotPasswordActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_forgot_password);

        final EditText editText = findViewById(R.id.reset_pw);

        final Button resetPw = findViewById(R.id.reset_pw_sign_up);
        resetPw.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                resetPassword(editText.getText().toString().trim());
            }
        });

    }


    private void resetPassword(String emailAddress){
        FirebaseAuth auth = FirebaseAuth.getInstance();

        auth.sendPasswordResetEmail(emailAddress)
                .addOnCompleteListener(new OnCompleteListener<Void>() {
                    @Override
                    public void onComplete(@NonNull Task<Void> task) {
                        if (task.isSuccessful()) {
                            Log.d(TAG, "Email sent.");

                            NavUtils.navigateUpFromSameTask(forgotPasswordActivity.this);
                            finish();
                        }else {
                            Toast.makeText(forgotPasswordActivity.this, "Could not reset Password", Toast.LENGTH_SHORT).show();
                        }
                    }
                });
    }
}

File: app/src/main/java/h_mal/appttude/com/driver/user/forgotPasswordActivity.kt
package h_mal.appttude.com.driver.user

import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.app.NavUtils
import com.google.android.gms.tasks.OnCompleteListener
import com.google.android.gms.tasks.Task
import com.google.firebase.auth.FirebaseAuth
import h_mal.appttude.com.driver.R


class forgotPasswordActivity : AppCompatActivity() {
    var TAG: String = "forgotPasswordActivity"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_forgot_password)
        val editText: EditText = findViewById(R.id.reset_pw)
        val resetPw: Button = findViewById(R.id.reset_pw_sign_up)
        resetPw.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View) {
                resetPassword(editText.text.toString().trim({ it <= ' ' }))
            }
        })
    }

    private fun resetPassword(emailAddress: String) {
        val auth: FirebaseAuth = FirebaseAuth.getInstance()
        auth.sendPasswordResetEmail(emailAddress)
            .addOnCompleteListener(object : OnCompleteListener<Void?> {
                override fun onComplete(task: Task<Void?>) {
                    if (task.isSuccessful) {
                        Log.d(TAG, "Email sent.")
                        NavUtils.navigateUpFromSameTask(this@forgotPasswordActivity)
                        finish()
                    } else {
                        Toast.makeText(
                            this@forgotPasswordActivity,
                            "Could not reset Password",
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                }
            })
    }
}